ansible/roles/eos/tasks/main.yml

```diff
@@ -41,34 +41,21 @@
   set_fact: properties_list="{{ configuration[hostname]['properties'] }}"
   when: configuration and configuration[hostname] and configuration[hostname]['properties'] is defined
 
-- name: copy boot-config
-  copy: src=boot-config
-        dest=/mnt/flash/boot-config
-
 - name: Expand {{ hostname }} properties into props
   set_fact: props="{{ configuration_properties[item] | combine(props | default({})) }}"
   with_items: properties_list
   when: hostname in configuration and configuration_properties[item] is defined
 
-- name: build a startup config
-  template: src="{{ topo }}-{{ props.swrole }}.j2"
-            dest=/mnt/flash/startup-config
-  when: hostname in configuration
-
-- name: Restart the box
-  command: /sbin/shutdown -r now "Ansible updates triggered"
-  when: hostname in configuration
-
-- name: Pause for reboot
-  pause: seconds=30
+- name: copy boot-config
+  copy: src=boot-config
+        dest=/mnt/flash/boot-config
   when: hostname in configuration
+  notify:
+    - Update VM state
```

Comment:
Should we wait until we update startup-config and then reboot?
I think we should remove notify here.
@nazariig  What is your opinion?

Reply:
@pavel-shirshov The handlers will be executed at the end of play only once, regardless of how many times they were triggered. It's by design.

> **Handlers: Running Operations On Change**
> Regardless of how many tasks notify a handler, it will run only once, after all of the tasks complete in a particular play.

The idea is to do config update, when either one of the conditions is met: the system will wait untill all the relevant configurations are updated and only then trigger the reboot sequence if necessary.

Reply:
Got it! Thank you for the explanations.

Reply:
@pavel-shirshov welcome.